### PR TITLE
fix: use xc table for at import

### PR DESCRIPTION
**Pull request by mertmit**
## Change Summary

closes: #8235 

- Use ai ID
- Restrict editing of system columns

## Change type

- [x] fix: (bug fix for the user, not a fix to a build script)

### CODE REVIEW

**Review by dstala — COMMENTED** [review]
- [ ] two fields with same name `Id`. scenario needs to be handled (known issue, as discussed during call)
Screenshot 2024-05-30 at 12 42 05 PM


- [ ] data import failed
Screenshot 2024-05-30 at 12 31 14 PM

**Comment by mertmit** [comment]
Data import part is a side-effect of system field update restrictions, I will think of a solution.

For duplicate id -> we should decide approach:
- Rename our fields
- Rename incoming fields

need your input on this @dstala

**Comment by dstala** [comment]
- Let ID field be a system keyword in NocoDB. No other field can have ID field
- Rename Airtable's field with title "ID" to "ID1".

File: packages/nc-gui/components/dlg/AirtableImport.vue [hunk 1]
```diff
@@ -403,22 +403,30 @@ function downloadLogs(filename: string) {
...
+            <a-tooltip placement="top">
+              <template #title>
+                <span>{{ $t('title.comingSoon') }}</span>
+              </template>
+              <a-checkbox v-model:checked="syncSource.details.options.syncUsers" disabled>
+                {{ $t('labels.importUsers') }}
+              </a-checkbox>
+            </a-tooltip>
+          </div>
+          -->
```
Comment (pranavxc) [inline packages/nc-gui/components/dlg/AirtableImport.vue:429]: Can be removed if no longer needed or going to enable in future.

File: packages/nocodb/src/db/BaseModelSqlv2.ts [hunk 2]
```diff
@@ -3218,6 +3218,12 @@ class BaseModelSqlv2 {
         { ignoreView: true, getHiddenColumn: true },
       );
 
+      if (!prevData) {
+        NcError.recordNotFound(
+          typeof id === 'string' ? id : JSON.stringify(id),
```
Comment (pranavxc) [inline packages/nocodb/src/db/BaseModelSqlv2.ts:3223]: If possible let's move the `stringify` part to `recordNotFound` method.

**Review by pranavxc — APPROVED** [review]

File: packages/nc-gui/components/dlg/AirtableImport.vue (hunk 1 above)
Comment (mertmit) [reply packages/nc-gui/components/dlg/AirtableImport.vue:429]: We've disabled for now as it is not functioning properly, there is a possibility to re-enable

**Review by pranavxc — APPROVED** [review]

**Pull request merged**